### DISCUSSION TITLE
[ Gift Card Help ] Can't process payment with 0 amount

### DISCUSSION THREAD

**Discussion by DidoMarchet** [saleor/saleor#8967] [General] [answered]
Hi,
if I use a gift card and the amount goes to 0, the payment can't be processed (Braintree gateway).

Do you have any advice about that?

Thanks in advance and regards,

Davide

**Comment by karolkielecki** [comment]
Hi @DidoMarchet. Which version are you running? It will be helpful if you share more details. Do you receive any errors?

**Comment by DidoMarchet** [comment]
@karolkielecki you're right sorry!

Well the Saleor version is: 3.1.0-a.0

The commerce uses SDK v0.3.0-10

It returns me this error:
```
[{"data": {"checkoutComplete": {"errors": [{"code": "PAYMENT_ERROR", "field": null, "message": "Unable to process transaction. Please try again in a moment", "__typename": "CheckoutError"}], "order": null, "confirmationNeeded": false, "confirmationData": null, "__typename": "CheckoutComplete"}}}]
```

If the problem is the old version of Saleor I can try to update it but I want to be sure there's no breaking changes with the old version of the SDK.

Thanks in advance and kind regards,

Davide

**Reply by karolkielecki** [reply]
Well, you can try to update to the latest 3.1. However, please bear in mind that it's still alpha. We even haven't updated our documentation regarding gift cards. We recommend using the latest 3.0 beta for new projects--but it doesn't support gift cards. Please let me know how gift cards work if you update to the latest alpha. Cheers

**Reply by DidoMarchet** [reply]
@karolkielecki Wi'll try and I let you know.

Regards,

Davide

**Reply by DidoMarchet** [reply]
@karolkielecki We've update to the last version 3.1!

**Reply by karolkielecki** [reply]
@DidoMarchet How does it work? AFAIK, the gift card feature is still under development. So, there can be some issues.

**Reply by DidoMarchet** [reply]
@karolkielecki 
Ok thanks! Unfurtunatly but I found this issues with discount too.
If I apply a discount it affect the sipping methods.
I've got a free shipping for orders > 75$.
If I appply the discount it affect the shipping method and it doesn't works because the amount  is 0.

Any advice?

Regards,

Davide

<img width="1404" alt="Screenshot 2022-01-20 at 13 03 43" src="https://user-images.githubusercontent.com/38940340/150336475-14432e2e-889e-4e2c-abc8-05d646f14c62.png">

**Comment by DidoMarchet** [comment, marked as answer]
@karolkielecki 
Hi! The right version installed is 3.1.0-a.0. 
Btw, you are aware about the  discount affecting shipping method and the related issue? Or is my fault?  


Thanks in advance and regards,

Davide

**Reply by karolkielecki** [reply]
Hi @DidoMarchet! Thanks for reporting. Let me check it with the team.

**Reply by DidoMarchet** [reply]
@karolkielecki  waiting for your answer, thanks and regards,
 
 Davide

**Reply by karolkielecki** [reply]
@DidoMarchet Yesterday, we merged two PRs fixing the issue described by you above. Check it out: 
- https://github.com/saleor/saleor/pull/8938
- https://github.com/saleor/saleor/pull/8979

**Reply by DidoMarchet** [reply]
@karolkielecki great thanks!

**Reply by DidoMarchet** [reply]
> @DidoMarchet Yesterday, we merged two PRs fixing the issue described by you above. Check it out: 
> 
> - https://github.com/saleor/saleor/pull/8938
> 
> - https://github.com/saleor/saleor/pull/8979

It works as aspected!

**Reply by karolkielecki** [reply]
Great! @DidoMarchet
